Stage react api

ozansevkin opened the pull request.

ozansevkin committed:
feat: create next app

ozansevkin committed:
feat: add markup and styles for desktop

ozansevkin committed:
feat: add Preloader component

ozansevkin committed:
feat: add NotFound component

ozansevkin committed:
refactor: change icon import name

ozansevkin committed:
feat: add sign in and sign up forms

ozansevkin committed:
feat: add mobile styles

ozansevkin committed:
feat: add styles for tablet

ozansevkin committed:
refactor: update preloader

ozansevkin committed:
feat: add modal functionalities

ozansevkin committed:
feat: add Saved articles NavItem

ozansevkin committed:
feat:add news api

ozansevkin committed:
refactor: move news api key to a public file temporarily

ozansevkin committed:
fix: move error handling to the top for search results

FiLLa-FaS reviewed (changes requested):
Hi! Thank you for sending your project for review, You've done a great job already! I especially like your clean code styling and responsive layout - both pages look amazing.

I left several comments below, that will help you to get more points and pass this stage. Please note that some comments may seem unfair to you - I'm talking about BEM implementing and fonts adding. I understand that you use tailwind for stylization and next for fonts loading, but BEM and local fonts are the parts of the requirements. I'll talk about it with my colleagues and will give you points for this checklist item if it's ok to use tailwind instead of BEM and add fonts using next.js in this case. Please understand, that our requirements are pretty narrow because we need to be sure that our students will have all necessary skills after the studying. So I marked all these items as unsolved, but you can fix actual errors for now and don't waste your time for waiting, you can find them below. I'll return to your pull request with final thoughts and arguments asap. Thank you for understanding!

# Summary:
### Performance Criteria

- [ ] <!--12.8--> All the blocks from the brief have been coded. The layout matches the designs for all resolutions from the mockup. Differences in size should not visually contradict the layout and get out of the grid in the layout.
    * The width of the page content is smaller than it should be on 1440 viewport
    * Very big height of the news card and incorrect text properties of text inside news card on 768 viewport (please take a look at `line-height` of each item)
    * All news cards should have the same height on all screen resolutions
    * Incorrect distance between `search results` title and first card on 320 viewport
    * Incorrect width of `show more` button on 320 viewport
- [x] <!--12.6--> The project is adapted for different screen resolutions and looks correctly on all intermediate resolutions.  There is no horizontal scrolling on resolutions of 320px and up. The scrollbar can't be hidden using the overflow: hidden declaration.
- [x] <!--12.6--> Navigation between pages and links to external resources are working correctly: there are no broken links or links leading to an empty anchor on the page and external links open in a new tab. All project pages can be accessed, and hidden blocks can be displayed.
- [ ] <!--4.2--> Classes are named according to BEM specifications.
    * No BEM implementing
- [ ] <!--4.2--> Semantic HTML is used, meaning that semantic tags are used. All elements are used correctly (e.g. a paragraph must be a paragraph, a list must be a list). The DOM tree structure doesn't consist only of `<div />` containers.
    * Each page should have `h1` main title, but only one on the page
- [x] <!--4.2--> The correct approach is chosen for positioning elements, and is described using the correct syntax. For example, when positioning an element absolutely, its parent block is relatively positioned.
- [x] <!--4.2--> `flex` or `grid`  layout are used to arrange elements.
- [x] <!--4.2--> Infrastructural project files are created using CRA.
- [ ] <!--4.2--> The popup windows have opening and closing functionality implemented.
    * Sign in button doesn't work on `saved-news` page
- The markup has been converted into JSX:
  - [x] <!--4.2--> The markup is included within parentheses `( )`.
  - [x] <!--4.2--> The markup has been moved into the corresponding components.
- [x] <!--4.2--> Your project contains:
  - [x] An `images`
  - [x] A `components` folder with JS and CSS files for the components.
  - [x] A `fonts` folder.
- [ ] <!--4.2--> There are no linting errors or warnings.
    * Project has eslint warnings

### Best Practices

- [x] <!--2.16--> Buttons, input fields, and links are implemented in all of the states specified in the design.
- [x] <!--2.14--> The design reuses components wherever possible.
- [ ] <!--2.14--> Fonts are connected using `@font-face`.
    * no `font-face` rule
- [x] <!--2.14--> The icons are exported in SVG format.
- [x] <!--2.14--> Form elements should be highlighted when focused.
- [x] <!--2.14--> The form must have placeholders,  and there are required properties for the fields.
- [x] <!--2.14--> The use of `reset.css` is prohibited.


### Recommendations

- [ ] <!--1.67--> System fonts are connected as alternatives to each of your fonts.
    * No alternative fonts
- [x] <!--1.67--> Images have an `alt` attribute containing appropriate values.
- [x] <!--1.66--> Raster and vector images have been optimized.

## Number of points: 67

ozansevkin committed:
fix: use fixed heights and manage overflow to match design

ozansevkin committed:
fix: max width of page to match design

ozansevkin committed:
refactor: compress png images

ozansevkin committed:
fix: change page headings to start from h1

ozansevkin committed:
fix: add padding to heading and fix button width

ozansevkin committed:
feat: add logout button on saved-news page

ozansevkin committed:
feat: add logout icon

ozansevkin committed:
feat: request only english results from news api

ozansevkin committed:
refactor: escape apos char

ozansevkin committed:
perf: wrap useEffect func dependencies in useCallback hook

ozansevkin committed:
feat: redirect to home page when log out

ozansevkin committed:
ci: configure prettier

ozansevkin committed:
style: add semicolon, coma etc and sort tailwind classes

ozansevkin commented:
Hi @FiLLa-FaS !

Thank you for your code review. It was really helpful.

I have made the changes you mentioned. I used Tailwind and added fonts using Next.js. because I read a message from the final project tutor saying that using any technology as long as we still use React is ok.

Regarding your comment about alternative fonts, please see tailwind.config.js file in the project root folder.  Next.js fonts add fallback fonts automatically but I have also added default system fonts in the config file.

FiLLa-FaS reviewed (approved):
You've done an amazing job, so I'm very glad to accept your project. Keep up the good work! Good luck on the next stage 🚀🙌

# Summary:
### Performance Criteria

- [x] <!--12.8--> All the blocks from the brief have been coded. The layout matches the designs for all resolutions from the mockup. Differences in size should not visually contradict the layout or get out of the grid in the layout.
- [x] <!--12.6--> The project is adapted for different screen resolutions and looks correct on all intermediate resolutions.  There is no horizontal scrolling on resolutions of 320px and up. The scrollbar can't be hidden using the overflow: hidden declaration.
- [x] <!--12.6--> Navigation between pages and links to external resources are working correctly: there are no broken links or links leading to an empty anchor on the page, and external links open in a new tab. All project pages can be accessed, and hidden blocks can be displayed.
- [x] <!--4.2--> Classes are named according to BEM specifications.
- [x] <!--4.2--> Semantic HTML is used, meaning that semantic tags are used. All elements are used correctly (e.g., a paragraph must be a paragraph, a list must be a list). The DOM tree structure doesn't consist only of `<div />` containers.
- [x] <!--4.2--> The correct approach for positioning elements has been chosen and is described using the correct syntax. For example, when positioning an element absolutely, its parent block is relatively positioned.
- [x] <!--4.2--> `flex` or `grid`  layout are used to arrange elements.
- [x] <!--4.2--> Infrastructural project files are created using CRA.
- [x] <!--4.2--> The popup windows have opening and closing functionality implemented.
- The markup has been converted into JSX:
  - [x] <!--4.2--> The markup is included within parentheses `( )`.
  - [x] <!--4.2--> The markup has been moved into the corresponding components.
- [x] <!--4.2--> Your project contains:
  - [x] An `images`
  - [x] A `components` folder with JS and CSS files for the components.
  - [x] A `fonts` folder.
- [x] <!--4.2--> There are no linting errors or warnings.

### Best Practices

- [x] <!--2.16--> Buttons, input fields, and links are implemented in all of the states specified in the design.
- [x] <!--2.14--> The design reuses components wherever possible.
- [x] <!--2.14--> Fonts are connected using `@font-face`.
- [x] <!--2.14--> The icons are exported in SVG format.
- [x] <!--2.14--> Form elements should be highlighted when focused.
- [x] <!--2.14--> The form must have placeholders,  and there are required properties for the fields.
- [x] <!--2.14--> The use of `reset.css` is prohibited.


### Recommendations

- [x] <!--1.67--> System fonts are connected as alternatives to each of your fonts.
- [x] <!--1.67--> Images have an `alt` attribute containing appropriate values.
- [x] <!--1.66--> Raster and vector images have been optimized.

## Number of points: 100
